### PR TITLE
[full-ci] Refactor storage configuration to make it similar to other oCIS configuration

### PR DESCRIPTION
**Pull request by C0rby**
[Read this issue for more details](https://github.com/owncloud/ocis/issues/3568)

The configuration and service commands now live in their own packages in the extensions directory. The configuration is missing the env variables and descriptions. They will be added in following PRs. Also there might be some more clean ups possible and some configuration could be added to the common struct but this can be done later as well.

### CODE REVIEW

**Comment by ownclouders** [comment]
:boom: Acceptance test [<strong>Web-Tests-ocis-smoke-ocis-storage-2</strong>](https://drone.owncloud.com/owncloud/ocis/10850/35/1) failed. Further test are cancelled...

**Comment by C0rby** [comment]
Lesson for next time: Open multiple smaller PRs instead of one big PR... :crying_cat_face:

File: ocis-pkg/ldap/ldap.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+const _caTimeout = 5
+
+func WaitForCA(log log.Logger, insecure bool, caCert string) error {
+	if !insecure && caCert != "" {
+		if _, err := os.Stat(caCert); errors.Is(err, os.ErrNotExist) {
+			log.Warn().Str("LDAP CACert", caCert).Msgf("File does not exist. Waiting %d seconds for it to appear.", _caTimeout)
+			time.Sleep(_caTimeout * time.Second)
+			if _, err := os.Stat(caCert); errors.Is(err, os.ErrNotExist) {
+				log.Warn().Str("LDAP CACert", caCert).Msgf("File does still not exist after Timeout")
```
Comment (phil-davis) [inline ocis-pkg/ldap/ldap.go:19]:
```suggestion
				log.Warn().Str("LDAP CACert", caCert).Msgf("File still does not exist after Timeout")
```

File: ocis/pkg/command/storagegateway.go [hunk 2]
```diff
@@ -14,11 +14,11 @@ func StorageGatewayCommand(cfg *config.Config) *cli.Command {
 		Usage:    "start storage gateway",
 		Category: "extensions",
 		//Flags:    flagset.GatewayWithConfig(cfg.Storage),
-		Before: func(ctx *cli.Context) error {
-			return ParseStorageCommon(ctx, cfg)
-		},
+		// Before: func(ctx *cli.Context) error {
+		// 	return ParseStorageCommon(ctx, cfg)
+		// },
```
Comment (phil-davis) [inline ocis/pkg/command/storagegateway.go:19]: delete?

**Review by phil-davis — COMMENTED** [review]
Just small things - looking good if CI will pass.

**Review by wkloucek — APPROVED** [review]
We're merging this for now and reenable the Parallel-Deployment testsuite in a separate PR, as we also add all the missing env variables and descriptions.

**Comment by wkloucek** [comment]
Follow up in https://github.com/owncloud/ocis/pull/3589

**Pull request merged**